### PR TITLE
Tutorial: Server components and agents for web3 apps

### LINKED ISSUE #13158: Suggest a tutorial: Server components for web3 apps
### Tutorial title

Server components for web3 apps

### Tutorial description

After this tutorial, you will be able to write TypeScript servers that listen to events on a blockchain and respond accordingly with their own transactions. This will enable you to write applications that are centralized (because the server is a point of failure), but can interact with web3 entities.

### Tutorial tags

server, offchain, centralized

### Skill level

Beginner

### Hosted on ethereum.org or hosted elsewhere?

Hosted on ethereum.org

### For tutorials to be hosted on Ethereum.org: Tutorial Content

- Introduction: Why a Centralized Server?
- Getting started: Node, TypeScript, and Viem
- Listening to Events
- Responding with Transactions
- Conclusion, when is this an appropriate solution?

### For tutorials hosted elsewhere: URL to tutorial

_No response_

### Would you like to work on this issue?

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by qbzzt**
A tutorial on how to write a centralized server component / agent 

## Description

After this tutorial, you will be able to write TypeScript servers that listen to events on a blockchain and respond accordingly with their own transactions. This will enable you to write centralized applications (because the server is a point of failure), but can interact with web3 entities. The same techniques can also be used to write an agent that responds to on-chain events without a human in the loop.

## Related Issue

- Closes #13158

### CODE REVIEW

File: public/content/developers/tutorials/server-components/fig-1.svg
Comment (wackerow) [inline public/content/developers/tutorials/server-components/fig-1.svg:1]: @nloureiro, any chance we could repaint these diagrams with our design system elements? Doesn't need to be fancy, just on theme with other diagrams ideally

**Comment by qbzzt** [comment]
Yes. Can you tell me what changes?

File: public/content/developers/tutorials/server-components/fig-1.svg
Comment (qbzzt) [reply public/content/developers/tutorials/server-components/fig-1.svg:1]: @corwintines has redone diagrams. Can you please review the text? I'm getting ready to submit my zero-knowledge article about secret state, and it includes a link to this article.

**Review by corwintines — APPROVED** [review]
LGTM, thanks @qbzzt

**Comment by qbzzt** [comment]
> LGTM, thanks @qbzzt

Thank you!

**Pull request merged**